### PR TITLE
[bitnami/nats] Add tests and publish using VIB

### PR DESCRIPTION
**Pull request by FraPazGal**
### Description of the change

The main objective of this PR is to publish our Bitnami NATS Helm Chart using VMware Image Builder. In order to do that, several changes are included:

- Adding the asset to the publishing workflow
- Increasing the existing test coverage of the asset by adding GOSS tests.
- Increasing the existing test coverage of the asset by adding Cypress tests.

### Benefits

- Ensuring higher quality of the Helm charts.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could show to be potentially flaky. 

### Applicable issues

NA

### Additional information

Tested in https://github.com/FraPazGal/charts/pull/84

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/nats/goss/goss.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  {{- $msg := printf "msg_%s" (randAlpha 5) }}
+  {{- $auth := printf "--user=%s --password=%s" .Vars.auth.user .Vars.auth.password }}
+  {{- $client_endpoint := printf "-s nats://nats:%d" .Vars.service.ports.client }}
+  manage-key:
+    ## We need to have the subscriber process running at the same time the publishing is done
+    ## The sleep is needed to avoid a race condition where the publishing is done before the subscriber is ready
+    ## The subscriber process needs to be killed so there is no dangling process making goss timeout
+    exec: ( USER=1001 nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
+    ## Goss seems to regard the exit status of this test as unfinished (-1) even though the stdout is correct and the subscriber process is stopped
+    exit-status: -1
```
Comment (joancafom) [inline .vib/nats/goss/goss.yaml:18]: I think you can avoid all of this if you use the `timeout` [command](https://man7.org/linux/man-pages/man1/timeout.1.html). You can try change it to something like this and see if it works:

```
$ (USER=1001 timeout 10 nats --user=test_nats_client --password='ComplicatedPassword123!4' sub -s nats://nats:4224 subj_wiggle || true) & &&
USER=1001 nats --user=test_nats_client --password='ComplicatedPassword123!4' pub -s nats://nats:4224 subj_wiggle here
15:40:06 Published 4 bytes to "subj_wiggle"
[#1] Received on "subj_wiggle"
here
```

File: .vib/nats/goss/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+command:
+  {{- $subject := printf "subj_%s" (randAlpha 5) }}
+  {{- $msg := printf "msg_%s" (randAlpha 5) }}
+  {{- $auth := printf "--user=%s --password=%s" .Vars.auth.user .Vars.auth.password }}
+  {{- $client_endpoint := printf "-s nats://nats:%d" .Vars.service.ports.client }}
+  manage-key:
+    ## We need to have the subscriber process running at the same time the publishing is done
+    ## The sleep is needed to avoid a race condition where the publishing is done before the subscriber is ready
+    ## The subscriber process needs to be killed so there is no dangling process making goss timeout
+    exec: ( USER=1001 nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
```
Comment (joancafom) [inline .vib/nats/goss/goss.yaml:16]: Use the same user as in securityContext

```suggestion
    exec: ( USER={{ .Vars.containerSecurityContext.runAsUser }} nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
```

File: .vib/nats/vib-verify.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
...
...F0ZWRQYXNzd29yZDEyMyE0CmNsdXN0ZXI6CiAgYXV0aDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIHVzZXI6IHRlc3RfbmF0c19jbHVzdGVyCiAgICBwYXNzd29yZDogQ2x1c3RlclBhc3N3b3JkMTIzITQKcG9kU2VjdXJpdHlDb250ZXh0OgogIGVuYWJsZWQ6IHRydWUKICBmc0dyb3VwOiAxMDAyCmNvbnRhaW5lclNlY3VyaXR5Q29udGV4dDoKICBlbmFibGVkOiB0cnVlCiAgcnVuQXNVc2VyOiAxMDAyCmNvbnRhaW5lclBvcnRzOgogIGNsaWVudDogNDIyNAogIGNsdXN0ZXI6IDYyMjYKICBtb25pdG9yaW5nOiA4MjI4CnNlcnZpY2U6CiAgcG9ydHM6CiAgICBtb25pdG9yaW5nOiA4MAogICAgY2xpZW50OiA0MjI0CiAgICBjbHVzdGVyOiA2MjI2CiAgdHlwZTogTG9hZEJhbGFuY2VyCnJlc291cmNlVHlwZTogc3RhdGVmdWxzZXQKcmVwbGljYUNvdW50OiAyCnN0YXRlZnVsc2V0OgogIHVwZGF0ZVN0cmF0ZWd5OiBSb2xsaW5nVXBkYXRlCmV4dHJhVm9sdW1lczoKICAtIG5hbWU6IG5hdHMtY2xpZW50CiAgICBlbXB0eURpcjoge30KZXh0cmFWb2x1bWVNb3VudHM6CiAgLSBuYW1lOiBuYXRzLWNsaWVudAogICAgbW91bnRQYXRoOiAvb3B0L2JpdG5hbWkvbmF0cy9iaW4vbmF0cwogICAgc3ViUGF0aDogbmF0cwppbml0Q29udGFpbmVyczoKICAtIG5hbWU6IG5hdHMtY2xpZW50CiAgICBpbWFnZTogZG9ja2VyLmlvL2JpdG5hbWkvZ29sYW5nCiAgICBpbWFnZVB1bGxQb2xpY3k6IEFsd2F5cwogICAgY29tbWFuZDoKICAgIC0gc2gKICAgIC0gLWMKICAgIC0gQ0dPX0VOQUJMRUQ9MCBnbyBpbnN0YWxsIGdpdGh1Yi5jb20vbmF0cy1pby9uYXRzY2xpL25hdHNAbGF0ZXN0CiAgICB2b2x1bWVNb3VudHM6CiAgICAtIG5hbWU6IG5hdHMtY2xpZW50CiAgICAgIG1vdW50UGF0aDogL2dvL2Jpbg==",
```
Comment (joancafom) [inline .vib/nats/vib-verify.json:25]: I don't think you need to set `statefulset.updateStrategy` here 😁

**Review by joancafom — COMMENTED** [review]
Thanks for your PR! I left you some comments, could you please take a look?

File: .vib/nats/goss/goss.yaml (hunk 1 above)
Comment (agomezmoron) [reply .vib/nats/goss/goss.yaml:18]: I do agree, it's smarter IMHO

Comment (FraPazGal) [reply .vib/nats/goss/goss.yaml]: It is indeed, changed 👍

**Review by agomezmoron — APPROVED** [review]
LGTM!

File: .vib/nats/goss/goss.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    mode: "0644"
+    filetype: file
+    exists: true
+    owner: root
+command:
+  {{- $running_user := printf "%d" .Vars.containerSecurityContext.runAsUser }}
+  {{- $auth := printf "--user=%s --password=%s" .Vars.auth.user .Vars.auth.password }}
+  {{- $client_port := printf "%d" .Vars.service.ports.client }}
+  {{- $local_endpoint := printf "-s nats://nats:%s" $client_port }}
+  local-key-management:
```
Comment (joancafom) [inline .vib/nats/goss/goss.yaml:12]: This is not really a _local_ key management, you are using the LoadBalancer... I think the naming could lead to some misinterpretation when reading the tests.

File: .vib/nats/goss/goss.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    exit-status: 0
+    stdout:
+    - /Received on.*{{ $subject }}/
+    - {{ $msg }}
+    timeout: 20000
+  {{- range $e, $i := until .Vars.replicaCount }}
+  cluster-key-management-{{ $i }}:
+    {{- $subject := printf "subj_%s" (randAlpha 5) }}
+    {{- $msg := printf "msg_%s" (randAlpha 5) }}
+    {{- $cluster_endpoint := printf "-s nats://nats-%d.nats-headless:%s" $i $client_port }}
```
Comment (joancafom) [inline .vib/nats/goss/goss.yaml:28]: Same here. This is not a `cluster` endpoint, but the individual access using the `headless` service

**Review by joancafom — APPROVED** [review]
The logic itself LGTM. I added some comments thought regarding the naming of the tests. Feel free to address them or not.

File: .vib/nats/goss/goss.yaml (hunk 4 above)
Comment (FraPazGal) [reply .vib/nats/goss/goss.yaml:12]: You are right, I overlooked this. Thanks for the catch!

**Pull request merged**